src/components/data-exploration/differential-expression-tool/DiffExprCompute.jsx

```diff
@@ -121,6 +121,54 @@ const DiffExprCompute = (props) => {
     );
   };
 
+  const renderError = () => {
+    const canRun = canRunDiffExpr();
+    if (!isFormValid) return <></>;
+    if (isVolcanoPlot && canRun !== canRunDiffExprResults.TRUE) {
+      return (
+        <Alert
+          message='Error'
+          type='error'
+          showIcon
+          description={(
+            <>
+              For the selected comparison, there are fewer than 3 samples with the minimum number of cells (10).
+              Volcano plot requires both p-values and logFC values, therefore the plot cannot be rendered.
+            </>
+          )}
+        />
+      );
+    } if (canRun === canRunDiffExprResults.INSUFFCIENT_CELLS_ERROR) {
+      return (
+        <Alert
+          message='Error'
+          description={(
+            <>
+              One or more of the selected samples/groups does not contain enough cells in the selected cell set.
+              Therefore, the analysis can not be run. Select other cell set(s) or samples/groups to compare.
+            </>
+          )}
+          type='error'
+          showIcon
+        />
+      );
+    }
+    if (canRun === canRunDiffExprResults.INSUFFICIENT_CELLS_WARNING) {
+      return (
+        <Alert
+          message='Warning'
+          description={(
+            <>
+              For the selected comparison, there are fewer than 3 samples with the minimum number of cells (10).
+              Only logFC values will be calculated and results should be used for exploratory purposes only.
+            </>
```

Comment:
also, all these 4 alerts are basically the same code with different description. I would have a variable for the description, set the value of description in each of these ifs and then return the Alert component with the parametrised description